lib/Scanner/ICAPClient.php

```diff
@@ -15,28 +15,29 @@ public function __construct(string $host, int $port) {
 		$this->port = $port;
 	}
 
-	private function connect(): bool {
-		$this->socket = \socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
-
-		if (!\socket_connect($this->socket, $this->host, $this->port)) {
-			return false;
-		}
-
-		# McAfee seems to not properly close the socket once all response bytes are sent to the client
-		# we use a 5 sec time out on receiving data
-		if (!\socket_set_option($this->socket, SOL_SOCKET, SO_RCVTIMEO, ["sec" => 5, "usec" => 0])) {
-			return false;
+	private function connect(): void {
+		$this->writeHandle = @\stream_socket_client(
+			"tcp://{$this->host}:{$this->port}",
+			$errorCode,
+			$errorMessage,
+			5
+		);
+
+		if (!$this->writeHandle) {
+			throw new InitException(
+				\sprintf(
+					'Cannot connect to "tcp://%s:%s": %s (code %d)',
+					$this->host,
+					$this->port,
+					$errorMessage,
+					$errorCode
+				)
+			);
 		}
-		return true;
 	}
 
 	private function disconnect(): void {
-		@\socket_shutdown($this->socket);
-		@\socket_close($this->socket);
-	}
-
-	public function getLastSocketError(): int {
-		return \socket_last_error($this->socket);
+		@\fclose($this->writeHandle);
```

Comment:
Not errors but warnings. This just bloat a log file with useless info. I don't see any practical usage of the fact the socket was closed with a warning in case we got a proper answer from the daemon.
And even the answer from the daemon is cut and broken - we handle that in our code. 
So the success of operation has no matter.

Reply:
Although it could be useless info, it would be something we should investigate at some point. No error in the log implies there is no bug, and if the app misbehaves for whatever reason we'll have problems if we need to investigate this area because we'll assume everything is ok when it might not.

Even a dumb warning such as "you're closing an already closed resource" could be a hint to detect potential performance problems above this code since we could be executing expensive operations twice or more times to reach here.
A comment would help to acknowledge that we're closing the resource multiple times (if that's the case), but if I don't see a comment saying that the server could have already closed the connection and we can get an error (I'm making up this, no idea if this could happen) I could assume this hasn't been taken care and that's something I could take into consideration if I need to debug this.